monarchmoney/monarchmoney.py

```diff
@@ -2352,14 +2392,54 @@ async def set_budget_amount(
             graphql_query=query,
         )
 
+    async def upload_and_parse_balance_history(
+        self,
+        account_id: str,
+        csv_content,
+        timeout: int = 300,
+        delay: int = 10,
+    ) -> bool:
+        """
+        Uploads and parses the balance history, updating the account balance on monarch money
+
+        :param account_id: The account ID to apply the history to.
+        :param csv_content: CSV representation of the balance history. Headers are Date, Amount, and Account Name.
```

Comment:
This should probably be something more structured than a str?  If there are specific columns and column types/formats needed, perhaps using a `List[Dict[str, Any]]`?  Alternatively, use either Python's built-in CSV parsing library or, maybe, Pandas.

nit: move the description of the CSV format required to the body of the docblock

Reply:
If we're going this far to normalize the input, I'd vote for making the list value a class (or DataClass/NamedTuple). This way the caller doesn't need to worry about what the magic key names are.

Reply:
I agree that a dataclass would help the caller with the proper input. I've refactor it to use a dataclass. A second review over this would be appreciated!

Reply:
Personally, I'm fine with providing the CSV as a string. This gives the user the most flexibility to use the tools of their choice to build the CSV. If we require a normalized input, then they'll have to use a CSV parser to parse the input, only for it to generate a CSV once again. 

Optionally, we could provide users with a CSV builder kind of class if we want to help them along. IMO this could be an improvement made outside of this PR. That's @hammem 's call :)